myhpi/polls/models.py

```diff
@@ -18,24 +22,50 @@ class PollList(BasePage):
     parent_page_types = [
         "core.RootPage",
     ]
-    subpage_types = ["Poll"]
+    subpage_types = ["MajorityVotePoll", "RankedChoicePoll"]
     max_count = 1
 
     def get_context(self, request, *args, **kwargs):
         context = super().get_context(request, *args, **kwargs)
-        context.setdefault("poll_list", self.get_children().exact_type(Poll))
+        context.setdefault(
+            "poll_list",
+            self.get_children().specific().live()
+        )
         return context
 
 
-class Poll(BasePage):
-    question = models.CharField(max_length=254)
+class BasePoll(BasePage):
     description = CustomMarkdownField()
     start_date = models.DateField()
     end_date = models.DateField()
-    max_allowed_answers = models.IntegerField(default=1)
     results_visible = models.BooleanField(default=False)
 
-    participants = models.ManyToManyField(User, related_name="polls")
+    already_voted = models.ManyToManyField(User, related_name="base_polls", blank=True)
+
+    parent_page_types = [
+        "PollList",
+    ]
+    subpage_types = []
+    is_creatable = False
+
+    def in_voting_period(self):
+        return self.start_date <= datetime.date.today() <= self.end_date
+
+    def can_vote(self, user):
+        return self.in_voting_period() and user not in self.already_voted.all()
```

Comment:
I would like the poll to also have a list of groups I can limit it to. Otherwise profs etc. can also vote (I think this should be different from visibility, because visibility may not limit POSTing responses)